tests/object.cpp

```diff
@@ -691,32 +701,34 @@ TEST_CASE("object") {
             {"string", "hello"s},
             {"data", "olleh"s},
             {"date", Timestamp(10, 20)},
-            {"object array", AnyVec{ AnyDict{{"value", INT64_C(20)}}, AnyDict{{"value", INT64_C(21)}} } },
+            {"object array", AnyVec{AnyDict{{"value", INT64_C(20)}}, AnyDict{{"value", INT64_C(21)}}}},
         };
-        Object obj = create(dict, false);
+        Object obj = create(dict);
 
-        auto token1 = obj.add_notification_callback([&](CollectionChangeSet, std::exception_ptr) {
+        auto obj_table = r->read_group().get_table("class_all types");
+        Results result(r, *obj_table);
+        auto token1 = result.add_notification_callback([&](CollectionChangeSet, std::exception_ptr) {
```

Comment:
This is the change to the tests which make them actually reproduce the problem. Object notifiers only report direct changes while results notifiers also report changes to any linked objects, so using a results notifier instead makes it fire if any of the objects in the list change. All of the other changes to the tests are just to adapt them to the new API.